app/src/Lib/Docker/DockerShell.php

```diff
@@ -0,0 +1,434 @@
+<?php declare(strict_types=1);
+
+/* Copyright (c) 2022 - Daniel Weise <daniel.weise@concepts-and-training.de> - Extended GPL, see LICENSE */
+
+namespace CaT\Doil\Lib\Docker;
+
+use CaT\Doil\Lib\SymfonyShell;
+
+class DockerShell implements Docker
+{
+    use SymfonyShell;
+
+    protected const SALT = "/usr/local/lib/doil/server/salt";
+    protected const PROXY = "/usr/local/lib/doil/server/proxy";
+    protected const MAIL = "/usr/local/lib/doil/server/mail";
+
+    protected array $systems = [
+        self::SALT,
+        self::PROXY,
+        self::MAIL
+    ];
+
+    public function startContainerByDockerCompose(string $path) : void
+    {
+        $this->startDoilSystemsIfNeeded();
+
+        if (in_array($path, $this->systems)) {
+            return;
+        }
+
+        $cmd = [
+            "docker-compose",
+            "-f",
+            $path . "/docker-compose.yml",
+            "up",
+            "-d"
+        ];
+
+        $this->runTTYQuiet($cmd);
+
+        // Sometimes this file will not be deleted automatically,
+        // so we have to do it manually on start up.
+        // If not, apache wouldn't start.
+        $this->executeQuietCommand(
+            $path,
+            basename($path),
+            "rm",
+            "/run/apache2/apache2.pid"
+        );
```

Comment:
Why would this file be created here? Looks like we are interfering with stuff we should not interfere with...

Reply:
The file will be removed in this section. Sometimes the apache server didn't start by supervisorctl depending on this artifact. That is the way arround in the moment.